docs/usage/UsageWithTypescript.md

```diff
@@ -267,6 +267,21 @@ const rootReducer = combineReducers({ ... });
 type RootState = ReturnType<typeof rootReducer>;
 ```
 
+Switching the type definition of `RootState` with Redux Toolkit example:
+
+```ts
+//instead of defining the reducers in the reducer field of configureStore, combine them here:
+const rootReducer = combineReducers({ counter: counterReducer });
+
+//then set rootReducer as the reducer object of configureStore
+const store = configureStore({
+  reducer: rootReducer,
+  middleware: [...],
```

Comment:
Middleware here should almost never be an array though - and if passed in as an array probably with `as const`. Could you switch this up with the `getDefaultMiddleware` notation?

Reply:
sure! would you be ok with this syntax?

```ts
middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(yourMiddleware),
```

Reply:
That would be fine :)

Reply:
i submitted the PR with the agreed changes! here the [request](https://github.com/reduxjs/redux/pull/4414)